packages/superset-ui-connection/src/callApi/parseResponse.ts

```diff
@@ -1,26 +1,37 @@
-import { ParseMethod, SupersetClientResponse } from '../types';
+import { ParseMethod, SupersetClientResponse, TextResponse, JsonResponse } from '../types';
 
-function rejectIfNotOkay(response: Response): Promise<Response> {
-  if (!response.ok) return Promise.reject<Response>(response);
-
-  return Promise.resolve<Response>(response);
-}
-
-export default function parseResponse(
+export default function parseResponse<T extends ParseMethod = 'json'>(
   apiPromise: Promise<Response>,
   parseMethod: ParseMethod = 'json',
```

Comment:
`parseMethod` defined as `ParseMethod` can be out of sync with `T`.

```ts
// This will return Promise<JsonResponse>
parseResponse<'json'>({ parseMethod: 'raw' });
```